lib/parse.js

```diff
@@ -77,17 +77,13 @@ var parseValues = function parseQueryStringValues(str, options) {
             val = options.strictNullHandling ? null : '';
         } else {
             key = options.decoder(part.slice(0, pos), defaults.decoder, charset, 'key');
-            val = options.decoder(part.slice(pos + 1), defaults.decoder, charset, 'value');
+            val = options.decoder(part.slice(pos + 1), defaults.decoder, charset, 'value', options.comma);
         }
 
         if (val && options.interpretNumericEntities && charset === 'iso-8859-1') {
             val = interpretNumericEntities(val);
         }
 
-        if (val && typeof val === 'string' && options.comma && val.indexOf(',') > -1) {
-            val = val.split(',');
```

Comment:
Removing this is a breaking change; it forces anyone using a custom decoder to add this logic where they previously didn't require it.

Reply:
i agree, i will look into it more to see what can be done